pxteditor/history.ts

```diff
@@ -243,11 +245,23 @@ export function createSnapshot(text: ScriptText) {
             }
         }
 
+        if (result[pxt.HISTORY_FILE]) {
```

Comment:
nit: I think you could reduce some duplication if you just deleted lines 248 - 253 and let it continue on to the identical code after the try catch (259-264).